types/index.d.ts

```diff
@@ -24,14 +24,14 @@ export interface ComponentTracker {
 }
 
 export interface LoadableOptions<T> {
-  ErrorComponent?: React.ComponentType;
-  LoadingComponent?: React.ComponentType;
+  ErrorComponent?: React.ComponentType<T & { error: any }>;
+  LoadingComponent?: React.ComponentType<T>;
   render?: (options: { loading: boolean, error: boolean, ownProps: T, Component: React.ComponentType<T> }) => React.ReactElement<T>;
```

Comment:
I've noticed that `error: boolean` should actually be `error: Error | null`. 

Would also be nice if render options were extracted.

```
export type LoadableRenderOptions<T = any> = {
  loading: boolean;
  error: Error | null;
  ownProps: T;
  Component: React.ComponentType<T>;
};
```